### PR TITLE
Update defaults for extra values to support lists

### PR DESCRIPTION
**Pull request by domgoodwin**
Currently if you specify extraEnvs, extraVolumes or extraVolumeMounts it breaks the deployment. Updated so field looks for lists and not strings.

### CODE REVIEW

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
related to #530

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by domgoodwin** [comment]
There was an issue with the initContainer check, so I merged in latest from master

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by domgoodwin** [comment]
Looking at the filebeat PR, would you suggest adding the same pattern - where if string use the old method if not parse the yaml?

File: elasticsearch/templates/statefulset.yaml [hunk 1]
```diff
@@ -128,7 +128,11 @@ spec:
         {{- end }}
 {{ end }}
       {{- if .Values.extraVolumes }}
```
Comment (jmlrt) [inline elasticsearch/templates/statefulset.yaml:130]: Could you add a comment similar to the one added in Filebeat PR (https://github.com/elastic/helm-charts/blob/master/filebeat/templates/daemonset.yaml#L93-L98) so we know that this is done for backward compatibility?

**Comment by jmlrt** [comment]
jenkins test this please

File: elasticsearch/templates/statefulset.yaml [hunk 2]
```diff
@@ -183,7 +187,11 @@ spec:
           {{- end }}
 {{ end }}
       {{- if .Values.extraInitContainers }}
-{{ tpl .Values.extraInitContainers . | indent 6 }}
+      {{- if eq "string" (printf "%T" .Values.extraInitContainers) }}
+{{ tpl .Values.extraInitContainers . | indent 8 }}
+      {{- else }}
+{{ toYaml .Values.extraInitContainers | indent 8 }}
```
Comment (jmlrt) [inline elasticsearch/templates/statefulset.yaml:193]:
```suggestion
{{ tpl .Values.extraInitContainers . | indent 6 }}
      {{- else }}
{{ toYaml .Values.extraInitContainers | indent 6 }}
```

This is what makes https://devops-ci.elastic.co/job/elastic+helm-charts+pull-request+template-testing/527/CHART=elasticsearch,label=docker&&virtual/console fail

**Comment by domgoodwin** [comment]
I've updated with the requested changes now

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by domgoodwin** [comment]
I think I've fixed the linting errors, the jenkins console log was hard to read honestly

**Comment by jmlrt** [comment]
jenlins test this please

**Comment by jmlrt** [comment]
backported to `7.7` branch in bcaaaad

**Pull request merged**